skyportal/tests/frontend/test_sources.py

```diff
@@ -55,29 +56,70 @@ def test_classifications(driver, user, taxonomy_token, public_group, public_sour
         ],
     }
 
+    tax_name = str(uuid.uuid4())
+    tax_version = "test0.1"
+
     status, data = api(
         'POST',
         'taxonomy',
         data={
-            'name': str(uuid.uuid4()),
+            'name': tax_name,
             'hierarchy': simple,
             'group_ids': [public_group.id],
-            'version': "test0.1",
+            'version': tax_version,
         },
         token=taxonomy_token,
     )
     assert status == 200
 
-    driver.get(f"/become_user/{user.id}")  # TODO decorator/context manager?
+    driver.get(f"/become_user/{user.id}")
     driver.get(f"/source/{public_source.id}")
     driver.wait_for_xpath(f'//div[text()="{public_source.id}"]')
-    taxonomy_button = driver.wait_for_xpath(f'//div[@id="tax-select"]')
-    taxonomy_button.click()
+    driver.wait_for_xpath_to_be_clickable('//div[@id="tax-select"]').click()
+    driver.wait_for_xpath_to_be_clickable(
+        f'//*[text()="{tax_name} ({tax_version})"]'
+    ).click()
+    driver.wait_for_xpath("//body").click()
+    class_input = driver.wait_for_xpath('//*[@id="classification"]')
+    driver.scroll_to_element_and_click(class_input)
+    class_input.send_keys("Symmetrical", Keys.ENTER)
+    driver.scroll_to_element_and_click(
+        driver.wait_for_xpath("//*[@id='classificationSubmitButton']")
+    )
+    # Notification
+    driver.wait_for_xpath("//*[text()='Classification saved']")
+    # Button at top of source page
+    driver.wait_for_xpath(
+        "//span[contains(@class, 'MuiButton-label') and text()='Symmetrical']"
+    )
+    # Classifications list entry
+    driver.wait_for_xpath(f"//i[text()='{tax_name}']")
+    prob_span = driver.wait_for_xpath("//*[contains(text(), '(P=1)')]")
+    classification_entry_div = prob_span.find_element_by_xpath(
+        ".."
+    ).find_element_by_xpath("..")
+    # Hover to raise delete button
+    ActionChains(driver).move_to_element(classification_entry_div).perform()
+    delete_button = driver.wait_for_xpath(
+        "//button[starts-with(@name, 'deleteClassificationButton')]"
```

Comment:
What happens if there are more than 1 classifications (like from a previous test run)? I think this XPATH is not specific enough to action on the specific delete button and may delete some other classification.

Reply:
@profjsb a new test source is created on each run, so any old classifications will be associated with a different source, not the one being used each time the test is run again